src/test/java/uk/nhs/digital/nhsconnect/lab/results/inbound/NhsAckProducerServiceTest.java

```diff
@@ -0,0 +1,237 @@
+        when(interchangeParsingException.getMessage()).thenReturn("Interchange could not be processed.");
+
+        String nhsAck = nhsAckProducerService.createNhsAckEdifact(
+                WorkflowId.PATHOLOGY_ACK, interchangeParsingException);
+
+        String expectedNhsAck = readFile(NHSACK_IRI_RESPONSE_PATH).replace("\n", "");
+
+        assertEquals(expectedNhsAck, nhsAck);
+    }
+
+    @Test
+    void testCreateNhsAckMessageErrorIRMResponse() throws IOException {
+        final var interchangeParsingException = mock(MessagesParsingException.class);
+        when(interchangeParsingException.getSender()).thenReturn(INTERCHANGE_SENDER);
+        when(interchangeParsingException.getRecipient()).thenReturn(INTERCHANGE_RECIPIENT);
+        when(interchangeParsingException.getInterchangeSequenceNumber()).thenReturn(INTERCHANGE_CONTROL_REFERENCE);
+        when(interchangeParsingException.getMessage()).thenReturn("Messages could not be extracted.");
+
+        String nhsAck = nhsAckProducerService.createNhsAckEdifact(
+                WorkflowId.PATHOLOGY_ACK, interchangeParsingException);
+
+        String expectedNhsAck = readFile(NHSACK_IRM_RESPONSE_PATH).replace("\n", "");
+
+        assertEquals(expectedNhsAck, nhsAck);
+    }
+
+    @Test
+    void testCreateNhsAckMessageForScreeningReport() throws IOException {
+        final var interchange = mock(Interchange.class);
+        when(interchange.getInterchangeHeader()).thenReturn(InterchangeHeader.builder()
+                .sender(INTERCHANGE_SENDER)
+                .recipient(INTERCHANGE_RECIPIENT)
+                .translationTime(FIXED_TIME)
+                .sequenceNumber(INTERCHANGE_CONTROL_REFERENCE)
+                .build());
+
+        Message message1 = mock(Message.class);
+        when(message1.getMessageHeader()).thenReturn(new MessageHeader().setSequenceNumber(MESSAGE_SEQUENCE_NUMBER_1));
+        messageProcessingResults.add(new Success(message1, null));
+
+        Message message2 = mock(Message.class);
+        when(message2.getMessageHeader()).thenReturn(new MessageHeader().setSequenceNumber(MESSAGE_SEQUENCE_NUMBER_2));
+        messageProcessingResults.add(new Success(message2, null));
+
+        Message message3 = mock(Message.class);
+        when(message3.getMessageHeader()).thenReturn(new MessageHeader().setSequenceNumber(MESSAGE_SEQUENCE_NUMBER_3));
+        messageProcessingResults.add(new Success(message3, null));
+
+        String nhsAck = nhsAckProducerService.createNhsAckEdifact(
+                WorkflowId.SCREENING_ACK, interchange, messageProcessingResults);
+
+        String expectedNhsAck = readFile(NHSACK_SCREENING_RESPONSE_PATH).replace("\n", "");
+
+        assertEquals(expectedNhsAck, nhsAck);
+    }
+
+    private String readFile(String path) throws IOException {
+        try (InputStream is = this.getClass().getResourceAsStream(path)) {
+            return IOUtils.toString(is, StandardCharsets.UTF_8);
+        }
```

Comment:
We can now use `Files.lines(Path)` to get a `Stream<String>` of the contents, line-by-line
```suggestion
        Path filePath = FileSystems.getDefault().getPath(path);
        return Files.lines(filePath).collect(Collectors.joining("\n"));
```
if you think that's clearer. Might need to tweak the paths though